rust/src/lib/nm/nm_dbus/connection/conn.rs

```diff
@@ -205,8 +205,23 @@ impl TryFrom<NmConnectionDbusOwnedValue> for NmConnection {
 }
 
 impl NmConnection {
-    pub fn iface_name(&self) -> Option<&str> {
-        _connection_inner_string_member!(self, iface_name)
+    pub fn iface_name(&self) -> Option<String> {
+        if let Some(name) = _connection_inner_string_member!(self, iface_name) {
+            Some(name.to_string())
+        } else if self.iface_type() == Some(&NmIfaceType::Vlan) {
+            // VLAN will use <parent>.<vlan_id> if not defined.
+            if let Some((Some(parent), Some(vlan_id))) = self
+                .vlan
+                .as_ref()
+                .map(|v| (v.parent.as_ref(), v.id.as_ref()))
+            {
+                Some(format!("{parent}.{vlan_id}"))
+            } else {
+                None
+            }
+        } else {
+            None
+        }
```

Comment:
You said in a past PR review that nm_dbus should stay a a simple D-Bus wrapper and polishing and smart stuff done at nm level. I think this applies here.

Reply:
Yes. Let me find another approach, but current approach is simplest one, maybe just me been lazy.